pkg/cloudprovider/huaweicloud/instances.go

```diff
@@ -100,9 +110,17 @@ func (i *Instances) NodeAddressesByProviderID(ctx context.Context, providerID st
 // InstanceID returns the cloud provider ID of the node with the specified NodeName.
 // Note that if the instance does not exist, we must return ("", cloudprovider.InstanceNotFound)
 // cloudprovider.InstanceNotFound should NOT be returned for instances that exist but are stopped/sleeping
+//
+// Based on: https://github.com/kubernetes/cloud-provider-openstack/blob/2493d93/pkg/cloudprovider/providers/openstack/openstack_instances.go#L162
 func (i *Instances) InstanceID(ctx context.Context, nodeName types.NodeName) (string, error) {
-	klog.Infof("InstanceID is called. input nodeName: %s", nodeName)
-	return "", cloudprovider.NotImplemented
+	srv, err := i.getServerByName(nodeName)
+	if err != nil {
+		if err == ErrNotFound {
+			return "", cloudprovider.InstanceNotFound
```

Comment:
I'd prefer adding one error log when returns an error.

Same as below.

Reply:
note that it gets eventually logged by the node controller, e.g.

```
E0702 07:24:11.537080   20857 node_controller.go:428] failed to get instance ID from cloud provider: instance not found
```

Personally I would find it a bit redundant in this case, but it does not cost me much effort to add it if you still prefer ;-)

Reply:
It's no big deal. No worries.